exp/api/remote/remote_api_test.go

```diff
@@ -295,3 +299,136 @@ func TestRemoteAPI_Write_WithHandler(t *testing.T) {
 		}
 	})
 }
+
+func TestSnappyDecodeMiddleware(t *testing.T) {
+	tLogger := slog.Default()
+
+	var actReq *writev2.Request
+	successH := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		b, err := io.ReadAll(r.Body)
+		if err != nil {
+			t.Fatalf("failed to read body: %v", err)
+		}
+		actReq = &writev2.Request{}
+		if err := proto.Unmarshal(b, actReq); err != nil {
```

Comment:
We have this beautiful `remote.API` for client and server, yet we don't test/use it in this test and we manually unmarshal, send remote write.

Shouldn't we use it?